### PR TITLE
[Enhancement] Add fuse select assign pass

### PR DESCRIPTION
**Pull request by grimoire**
```python
    def forward(self, x: torch.Tensor):
        z = x / 2
        y = torch.zeros_like(x)
        y[x < 0.5] = z[x < 0.5]
        y[x > 0.2] = z[x > 0.2]
        return y
```

from:
origin onnx

to:
tmp onnx

### CODE REVIEW

**Comment by lvhan028** [comment]
Please rebase master since #594 reorganize `csrc`

**Comment by lvhan028** [comment]
I found the default value of `optimize` in `export` is `False`. Are we going to change it to `True` in the next release?

File: csrc/mmdeploy/backend_ops/torchscript/optimizer/passes/onnx/fuse_select_assign.cpp [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+  Graph pattern;
+  std::unordered_map<std::string, Value*> vmap;
+  torch::jit::parseIR(pattern_str, &pattern, vmap);
+
+  SubgraphMatcher matcher(pattern, MatchAttribute::NO_MATCH);
+  FuseSelectAssign(graph->block(), params, vmap, matcher);
+  torch::jit::EliminateDeadCode(
+      graph->block(), true,
+      torch::jit::DCESideEffectPolicy::ALLOW_DELETING_NODES_WITH_SIDE_EFFECTS);
```
Comment (lvhan028) [inline csrc/mmdeploy/backend_ops/torchscript/optimizer/passes/onnx/fuse_select_assign.cpp:145]: Will `ALLOW_DELETING_NODES_WITH_SIDE_EFFECTS` be safe here especially condering there is `DONT_DELETE_NODES_WITH_SIDE_EFFECTS`?

Comment (grimoire) [reply csrc/mmdeploy/backend_ops/torchscript/optimizer/passes/onnx/fuse_select_assign.cpp:145]: This pass is only used in ONNX Graph. ONNX Graph is SSA, theoretically, there are NO nodes with side effects that exist.

**Review by lvhan028 — APPROVED** [review]

**Comment by tpoisonooo** [comment]
Can you give an exmaple for this case ?

**Comment by grimoire** [comment]
@tpoisonooo 

```python
import torch
from mmdeploy.core import RewriterContext


# generate model and inputs
class TestModel(torch.nn.Module):

    def __init__(self) -> None:
        super().__init__()

    def forward(self, x, y, z):
        mask = z < 0.5
        x[mask] = y[mask]
        return x


model = TestModel()
x = torch.rand(1, 4, 8, 8)
y = torch.rand(1, 4, 8, 8)
z = torch.rand(1, 4, 8, 8)

# export model without pass
old_path = 'old.onnx'
torch.onnx.export(
    model, (x, y, z),
    old_path,
    input_names=['x', 'y', 'z'],
    output_names=['output'],
    dynamic_axes=dict(
        x={
            2: 'h',
            3: 'w'
        }, y={
            2: 'h',
            3: 'w'
        }, z={
            2: 'h',
            3: 'w'
        }),
    opset_version=11)

# export model with pass

from mmdeploy.apis.onnx.optimizer import model_to_graph__custom_optimizer  # noqa
from mmdeploy.backend.torchscript import ts_optimizer


def optimize_onnx(graph, params_dict, torch_out):
    ts_optimizer.onnx._jit_pass_fuse_select_assign(graph, params_dict)
    return graph, params_dict, torch_out


new_path = 'new.onnx'
with RewriterContext({}, onnx_custom_passes=optimize_onnx):
    torch.onnx.export(
        model, (x, y, z),
        new_path,
        input_names=['x', 'y', 'z'],
        output_names=['output'],
        dynamic_axes=dict(
            x={
                2: 'h',
                3: 'w'
            }, y={
                2: 'h',
                3: 'w'
            }, z={
                2: 'h',
                3: 'w'
            }),
        opset_version=11)

```

**Review by tpoisonooo — APPROVED** [review]
It seems helpful for postprocess,  LGTM.

**Pull request merged**